File: library/src/main/java/com/qingmei2/rximagepicker/config/sources/Gallery.java
package com.qingmei2.rximagepicker.config.sources;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * This annotation will be marked open Gallery，it will conflict with {@link Camera}
 * <p>
 * Created by qingmei2 on 2018/1/13.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface Gallery {
}

File: rximagepicker/src/main/java/com/qingmei2/rximagepicker/entity/sources/Gallery.kt
package com.qingmei2.rximagepicker.entity.sources

import androidx.annotation.IdRes
import com.qingmei2.rximagepicker.ui.gallery.BasicGalleryFragment
import kotlin.reflect.KClass

/**
 * This annotation will be marked open Gallery，it will conflict with [Camera]
 */
@Retention
@Target(
        AnnotationTarget.FUNCTION,
        AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER
)
annotation class Gallery(

        /**
         * the UI component that will be displayed.
         */
        val componentClazz: KClass<*> = BasicGalleryFragment::class,

        /**
         * the UI component type, as [Fragment] or [Activity].
         *
         * If you use custom UI component by [Activity], then set this value false.
         */
        val openAsFragment: Boolean = true,

        /**
         * the [Fragment] container
         */
        @IdRes val containerViewId: Int = 0
)
